### PR TITLE
logging: adds DigitalOcean Spaces logging endpoint support

### PR DESCRIPTION
**Pull request by mccurdyc**
## Relevant (Referenced) Link(s)

* [Fastly API Docs](https://docs.fastly.com/api/logging#api-section-logging_digitalocean)
* [fastly/go-fastly](https://github.com/fastly/go-fastly/blob/c2c2c62210800daab641161412fbe210004aac0a/fastly/digitalocean.go)

## Validation

```
$ make test
```

```
git clone git@github.com:mccurdyc/cli.git /tmp/cli && \
(
    cd /tmp/cli && \
    git checkout mccurdyc/logging-digitalocean && \
    make test && \
    rm -rf /tmp/cli \
)
```

### CODE REVIEW

File: pkg/logging/digitalocean/digitalocean_test.go [hunk 1]
```diff
@@ -0,0 +1,489 @@
...
+func deleteDigitalOceanOK(i *fastly.DeleteDigitalOceanInput) error {
+	return nil
+}
+
+func deleteDigitalOceanError(i *fastly.DeleteDigitalOceanInput) error {
+	return errTest
+}
+
+// pgpPublicKey returns a PEM encoded PGP public key suitable for testing.
+func pgpPublicKey() string {
```
Comment (mccurdyc) [inline pkg/logging/digitalocean/digitalocean_test.go:458]: This is repeated in quite a few endpoints. However, I don't think that we should create an exported function for it in testutil.

**Comment by pteichman** [comment]
Conflict fix is at https://github.com/fastly/cli/commit/df40a5fe1d4ba8a688a21b8910b02479f3acf99a

**Comment by mccurdyc** [comment]
Force-pushed Peter's commit.

**Comment by mccurdyc** [comment]
Force-pushed new factory pattern and strengthened tests commits squashed into a single commit.

**Comment by mccurdyc** [comment]
Force-pushed a squashed commit addressing CI failure.

**Comment by mccurdyc** [comment]
Force-pushed squashed commit for staticcheck check fix.

(Should run these checks locally...)

File: pkg/logging/digitalocean/create.go [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+	Placement         common.OptionalString
+	PublicKey         common.OptionalString
+}
+
+// NewCreateCommand returns a usable command registered under the parent.
+func NewCreateCommand(parent common.Registerer, globals *config.Data) *CreateCommand {
+	var c CreateCommand
+	c.Globals = globals
+	c.manifest.File.Read(manifest.Filename)
+	c.CmdClause = parent.Command("create", "Create a DigitalOcean logging endpoint on a Fastly service version").Alias("add")
```
Comment (phamann) [inline pkg/logging/digitalocean/create.go:45]: We should make sure we are consistent across all of these descriptions and call it "DigitalOcean Spaces" which our docs also do.

```suggestion
	c.CmdClause = parent.Command("create", "Create a DigitalOcean Spaces logging endpoint on a Fastly service version").Alias("add")
```

**Comment by mccurdyc** [comment]
Force-pushed rebasing master.

**Comment by mccurdyc** [comment]
Force-pushed addressing @phamann's review comment

`find . -type f -exec sed -i 's/DigitalOcean logging/DigitalOcean Spaces logging/g' {} \;`

File: pkg/logging/digitalocean/list.go [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+	manifest manifest.Data
+	Input    fastly.ListDigitalOceansInput
+}
+
+// NewListCommand returns a usable command registered under the parent.
+func NewListCommand(parent common.Registerer, globals *config.Data) *ListCommand {
+	var c ListCommand
+	c.Globals = globals
+	c.manifest.File.Read(manifest.Filename)
+	c.CmdClause = parent.Command("list", "List DigitalOcean endpoints on a Fastly service version")
```
Comment (phamann) [inline pkg/logging/digitalocean/list.go:27]:
```suggestion
	c.CmdClause = parent.Command("list", "List DigitalOcean Spaces endpoints on a Fastly service version")
```

**Comment by mccurdyc** [comment]
Force-pushed addressing @phamann's review comment.

(Well, at least you can see why my previous "fix" wasn't a fix) :)

Sorry for not catching that.

**Pull request merged**